azureLogIngestion/opentelemetry/index.ts

```diff
@@ -96,7 +98,7 @@ export default class OpenTelemetryAdapter {
                     : "https://otlp.nr-data.net:4317/v1/traces",
         })
 
-        this.traceProvider = new BasicTracerProvider({
+        this.traceProvider = new NRTracerProvider({
```

Comment:
The problem here is going to be that we're setting the resource before the first span, when we don't have an actual resource yet. 

What happens when you invoke .updateResource from within addSpan?

Reply:
You will need to look at the implementation of `BasicTracerProvider`, `Tracer` and `Span` to understand how this works. In short, when you start a Tracer, a bunch of meta data, including the resource, get passed to the Tracer from the TracerProvider, which in turn gets passed to the Span when a span is started. We get a Tracer instance in `addSpan`, then start/end a span. Whatever resource is set in the tracer provider is used and added to the batch.

Reply:
Yeah, I've been digging in there. Both the `Tracer` and `BasicTracerProvider` have private readonly `_resource` properties with a `resource` accessor.  I haven't yet figured out a way of overriding those that doesn't throw a build error.